Small fixes #10

ThePhoenix08 opened the pull request:
Summary of all changes
- Added "*.sqlite" to .gitignore to ignore the generated sqlite file in development
- Routing now is defaulted to provide <navbar route footer> three blocks structure unless if mentioned in routeObj to not to, this reduces redundancy as it's no longer needed to import and place navbar footer in each and every Page
- routeObj now has a new nullable bool flag "notShowNavbarFooter" to mention a specific route doesn't need navbar footer
- app.tsx now has an app div to it, it wraps the whole app, provides background gradient as its styling thus consistent background throughout without redundancy of placing a position absolute div on every page
- some semantic changes like footer tag, nav tag, header tag
- navbar is now sticky to the top, navbar stays accessible even after scrolling now
- fixed "emailto" to "mailto:", thus now it effectively triggers mail dialog when clicked
- added lucide icons in mobile mode to improve responsiveness of footer
- added useMediaQuery custom hook to check if the app is rendered on a small mobile screen or not
- Added asChild attribute to Buttons and added anchor tags as children to Buttons
- Removed Dialog Description as its primitive is paragraph tag, console error "p tag can't have div as child" fixed

Relevant issue: #10 

So, these are the changes this PR adds to the project. They contribute towards overall semantic improvements. I have manually tested the outputs of the changes, and the results are stable. I am doing this PR under KWoC. I request the mentor to kindly review this PR.

ThePhoenix08 committed:
add sqlite to gitignore

ThePhoenix08 committed:
modify routing to provide footer and navbar byDefault

ThePhoenix08 committed:
add app div, semantic changes in buttons, dialogs

ThePhoenix08 committed:
made footer responsive, add lucide icons, useMediaQuery hook

ThePhoenix08 committed:
fix: moved flag to redirect page route

mendacium-a11y commented:
- pls add some comments to additions wherever applicable
- add some screenshots of the ui changes for various screen sizes
- resolve the merge conflicts

ThePhoenix08 committed:
Merge branch 'aether' into small-fixes

ThePhoenix08 committed:
added comments explaining changes

ThePhoenix08 commented:
Screenshot of footer on mobile screen. Added lucide icons replacing the text links on phones. Icons work as links too.
Added comments wherever changes are made, explaining the changes in the latest commit { https://github.com/mendacium-a11y/Shortify/pull/12/commits/cb70be5f3f0dcfb4ed9908f81b356b6a81d042e8 }
![localhost_5173_(iPhone SE)](https://github.com/user-attachments/assets/f2e51c30-1c3d-4a74-b7cf-937e0db290a8)
